### PR TITLE
example fix for SUPPS-686 (TR-885)

### PR DESCRIPTION
**Pull request by vladm3**
added forceCompatMode flashVar to passthrough IE compat meta tag

### CODE REVIEW

File: modules/KalturaSupport/kalturaIframeClass.php [hunk 1]
```diff
@@ -578,6 +578,17 @@ private function getEntryTitle(){
 		return "Kaltura Embed Player iFrame";
 	}
 
+	function outputCompatibilityTags(){
+		$flashvars = $this->request->getFlashVars();
```
Comment (itaykinnrot) [inline modules/KalturaSupport/kalturaIframeClass.php]: please add check to the request (not null) just to be on the safe side @vladm3 
after that we can merge it - Thanks!

**Comment by vladm3** [comment]
@itaykinnrot Added the check to the request.

File: modules/KalturaSupport/kalturaIframeClass.php [hunk 2]
```diff
@@ -578,6 +578,19 @@ private function getEntryTitle(){
 		return "Kaltura Embed Player iFrame";
 	}
 
+	function outputCompatibilityTags(){
+		if (isset($this->request) && !empty($this->request)) {
```
Comment (OrenMe) [inline modules/KalturaSupport/kalturaIframeClass.php]: @vladm3 - any reason for the added security check here?
In general we have an IoC that injects the request into the kalturaIframeClass, so unless you saw it's necessary please remove the redundant check.

File: modules/KalturaSupport/kalturaIframeClass.php [hunk 3]
```diff
@@ -578,6 +578,19 @@ private function getEntryTitle(){
...
+			$flashvars = $this->request->getFlashVars();
+			if (isset($flashvars) &&
```
Comment (OrenMe) [inline modules/KalturaSupport/kalturaIframeClass.php]: @vladm3 the same goes for here, getFlashVars method always return an array(or a default value if you supply one), and if you look closely you will see that a few lines above where you call this method("outputCompatibilityTags") we also call "$this->request->getFlashVars()" so it is available to you also from there.

File: modules/KalturaSupport/kalturaIframeClass.php [hunk 4]
```diff
@@ -578,6 +578,19 @@ private function getEntryTitle(){
...
+					isset($flashvars['forceCompatMode']) &&
```
Comment (OrenMe) [inline modules/KalturaSupport/kalturaIframeClass.php]: in general "!empty" is the same as checking if it isset as well, and even if it's not set you won't get a warning from PHP on empty check on an undeclared var. AFAIK.

**Comment by eitanavgil** [comment]
@OrenMe  can you please take this with your team?

**Comment by vladm3** [comment]
@OrenMe thanks for the clues. I made some code rearranging and removed all redundant checks.

**Pull request merged**